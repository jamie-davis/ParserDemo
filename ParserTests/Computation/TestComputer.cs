using Xunit;
using Parser;
using TestConsoleLib;
using TestConsoleLib.Testing;
using System.Linq;
using Parser.Computation;

namespace ParserTests.Computation
{
    public class TestComputer
    {
        [Fact]
        public void ResultIsComputed()
        {
            //Arrange
            var testStrings = new [] {
                "5 + 3 * 6",
                "5 * 3 + 6",
                "5 * 2 + 3 / 4",
                "5 * (2 + 3) / 4",
            };

            //Act
            var results = testStrings
                .Select(s => new { String = s, Result = ArithmeticParser.Parse(s)})
                .ToList();

            //Assert
            var output = new Output();
            output.FormatTable(results.Select(r => new { r.String,  Description = r.Result.Describe(), Result = Computer.Compute(r.Result)}));
            output.Report.Verify();
        }

    }
}