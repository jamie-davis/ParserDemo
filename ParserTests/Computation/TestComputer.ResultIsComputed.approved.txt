String          Description                                      Result                        
--------------- --------------------- ---------------------------------                        
5 + 3 * 6       +[5, *[3, 6]]                                     23.00                        
5 * 3 + 6       *[5, +[3, 6]]                                     21.00                        
5 * 2 + 3 / 4   *[5, +[2, /[3, 4]]]                               10.75                        
5 * (2 + 3) / 4 *[5, /[(+[2, 3]), 4]]                              6.25                        
                                                                                               